Services/TMS/TrainingSearch/classes/Helper.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+class Helper {
+	const UNLIMITED_MEMBER_SPOTS = "&infin;";
+
+	/**
+	 * @var ilObjUser
+	 */
+	protected $g_user;
+
+	public function __construct() {
+		global $DIC;
+		$this->g_user = $DIC->user();
+	}
+
+	/**
+	 * Get needed values from bkm. Just best for user
+	 *
+	 * @param ilObjBookingModalities[] 	$bms
+	 * @param ilDateTime 	$crs_start_date
+	 *
+	 * @return array<intger, int | ilDateTime | string>
+	 */
+	public function getBestBkmValues(array $bkms, ilDateTime $crs_start_date) {
+		$first = array_shift($bkms);
+		$max_member = $first->getMember()->getMax();
+		$min_member = $first->getMember()->getMin();
+		$booking_start = $first->getBooking()->getBeginning();
+		$booking_end = $first->getBooking()->getDeadline();
+		$waiting_list = $first->getWaitinglist()->getModus();
+
+		foreach ($bkms as $key => $bkm) {
+			$max_member = $bkm->getMember()->compareMax($max_member);
+			$min_member = $bkm->getMember()->compareMin($min_member);
+			$booking_start = $bkm->getBooking()->compareBebeginning($booking_start);
+			$booking_end = $bkm->getBooking()->compareDeadline($booking_end);
+			$waiting_list = $first->getWaitinglist()->compareWaitingList($waiting_list);
+		}
+
+		$booking_start_date = clone $crs_start_date;
+		$booking_start_date->increment(ilDateTime::DAY, -1 * $booking_start);
+
+		$booking_end_date = clone $crs_start_date;
+		$booking_end_date->increment(ilDateTime::DAY, -1 * $booking_end);
+
+		$bookings_available = self::UNLIMITED_MEMBER_SPOTS;
+		if($max_member) {
+			$bookings_available = $max_member - $crs_member_count;
+		}
+
+		return array($max_member, $booking_start_date, $booking_end_date, $waiting_list, $min_member, $bookings_available);
+	}
+
+	/**
+	 * Get information about selected venue
+	 *
+	 * @param int 	$crs_id
+	 *
+	 * @return string[]
+	 */
+	public function getVenueInfos($crs_id) {
+		$vplug = ilPluginAdmin::getPluginObjectById('venues');
+		$vactions = $vplug->getActions();
+		$vassignment = $vactions->getAssignment((int)$crs_id);
+
+		if($vassignment) {
+			if($vassignment->isCustomAssignment()) {
+				$venue_id = -1;
+				$city = $vassignment->getVenueText();
+				$address = "";
+			}
+
+			if($vassignment->isListAssignment()) {
+				$venue_id = $vassignment->getVenueId();
+				$venue = $vactions->getVenue($venue_id);
+				$city = $venue->getAddress()->getCity();
+				$address = $venue->getAddress()->getAddress1();
+			}
+		}
+
+		return array($venue_id, $city, $address);
+	}
+
+	/**
+	 * Get information about selected provider
+	 *
+	 * @param int 	$crs_id
+	 *
+	 * @return string[]
+	 */
+	public function getProviderInfos($crs_id) {
+		$vplug = ilPluginAdmin::getPluginObjectById('trainingprovider');
+		$pactions = $vplug->getActions();
+		$passignment = $pactions->getAssignment((int)$crs_id);
+		$provider_id = -1;
+
+		if($passignment) {
+			if($passignment->isListAssignment()) {
+				$provider_id = $passignment->getProviderId();
+			}
+		}
+
+		return array($provider_id);
```

Comment:
Vielleicht an das Provider-Plugin?

Reply:
Verspricht das nicht auch 'nen bisschen viel? Das hier scheinen mit keine Infos zu sein, sondern nur 'ne Id.

Reply:
Ja im Moment verspricht das noch etwas zu viel. Ich hab den Namen aber extra so gewählt, weil ich davon ausgehen, dass wir in Zukunft noch mehr brauchen.